### PR TITLE
Provide a deprecation horizion

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: lib/dor/services/client/object.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Client
       # API calls that are about a repository object
       class Object < VersionedService
         extend Deprecation
+        self.deprecation_horizon = 'dor-service-client version 2.0'
```
Comment (ndushay) [inline lib/dor/services/client/object.rb]: isn't this dor-services-client ?

Comment (mjgiarlo) [reply lib/dor/services/client/object.rb]: It is. I think this is telling upstream codebases that when dsc 2 comes out, affected methods will go away.

Comment (ndushay) [reply lib/dor/services/client/object.rb]: dor-serviceSSSSS-client

Comment (mjgiarlo) [reply lib/dor/services/client/object.rb]: 😆

Yes, totally a typo there!

**Comment by jcoyne** [comment]
@ndushay typo fixed.

**Pull request merged**